polyply/src/build_system.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Processor for building systems with more than one molecule
+"""
+# this processor is not a subclass to the regular processor
+# class because it cannot run on a single molecule but needs
+# the system information
+import itertools
+import numpy as np
+from tqdm import tqdm
+from .random_walk import RandomWalk
+from .linalg_functions import u_vect
+from .topology import lorentz_berthelot_rule
+from .linalg_functions import norm_sphere
+
+def _compute_box_size(topology, density):
+    total_mass = 0
+    for molecule in topology.molecules:
+        for node in molecule.nodes:
+            if 'mass' in molecule.nodes[node]:
+                total_mass += molecule.nodes[node]['mass']
+            else:
+                total_mass += topology.atom_types['mass']
+    print(total_mass)
+    # amu -> kg and cm3 -> nm3
+    #conversion = 1.6605410*10**-27 * 10**27
+    box = (total_mass*1.6605410/density)**(1/3.)
+    return box
+
+
+def _prepare_topology(topology):
+    n_atoms = 0
+    for molecule in topology.molecules:
+         for node in molecule.nodes:
+            n_atoms+=1
+
+    # we need globally positions and nodes
+    positions = np.ones((n_atoms,3)) * np.inf
```

Comment:
```suggestion
    positions = np.full((n_atoms, 3), np.inf)
```
Or nan instead of inf?

Reply:
obsolete but we still stick with np.inf